src/opensynth/datasets/low_carbon_london/preprocess_lcl.py

```diff
@@ -117,69 +154,97 @@ def _get_settlement_offset(minute_value):
     return df_out
 
 
-def drop_dupes_and_replace_nulls(df: pd.DataFrame) -> pd.DataFrame:
+def drop_dupes_and_nulls(
+    df: pd.DataFrame, drop_nulls: bool = True
+) -> pd.DataFrame:
     """
     Function to drop duplicated readings and replace missing readings with 0.0
 
     Args:
         df (pd.DataFrame): Input dataframe
-
+        drop_nulls (bool): Whether to drop rows with NaN kwh values. If False,
+            will replace NaN kwh values with 0.0
     Returns:
         pd.DataFrame: Output dataframe
     """
-    logger.info("🗑 Dropping dupes and filling nulls with 0")
+    logger.info("🗑 Dropping dupes and nulls")
     df_out = df.copy()
     df_out = df_out.sort_values(
-        by=["LCLid", "date", "settlement_period"], ascending=True
+        by=["ID", "date", "settlement_period"], ascending=True
     )
     df_out = df_out.drop_duplicates(
-        subset=["LCLid", "date", "settlement_period"], keep="last"
+        subset=["ID", "date", "settlement_period"], keep="last"
     )
-    df_out["kwh"] = df_out["kwh"].replace("Null", np.float64())
-    df_out["kwh"] = df_out["kwh"].astype(float)
+    if drop_nulls:
+        df_out = df_out.dropna(subset="kwh")
+    else:
+        df_out["kwh"] = df_out["kwh"].fillna(0.0)
```

Comment:
Maybe add a log printing that nulls are filled with zero? I don't think it's obvious that drop_nulls = False, would mean nulls would be filled with zero. I know it says in the doc string but might as well add a simple print statement